### ISSUE TITLE
Deleting objects while in edit mode and then discarding changes causes the deleted objects to return

### ISSUE THREAD

**Issue by ozyx** [nasa/openmct#5206]
#### Summary

While in edit mode for a domain object, deleting other objects in the tree and then discarding changes causes the deleted objects to return.

#### Expected vs Current Behavior

The object should stay deleted since it bears no relation to the object being edited.

#### Steps to Reproduce

1. Create an object such as a Clock
2. Create an object which is editable, such as a Telemetry Table
3. While in edit mode for the Telemetry Table, delete the Clock from the object tree
4. Hit the 'X' button to discard changes
5. Click on 'My Items' or refresh the page
6. See that the Clock still persists after deletion

#### Environment

* Open MCT Version: 2.0.4
* Deployment Type: npm dev
* OS: Any
* Browser: Any

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?

**Comment by unlikelyzero** [comment]
@ozyx is this considered a regression?

**Comment by ozyx** [comment]
> @ozyx is this considered a regression?

I'm not sure, but I don't think so. I found this *really* old issue that seems somewhat related, at least in order of operations: [#368](https://github.com/nasa/openmct/issues/368)

If you replicate the steps in current openMCT, you can see the same behavior (substituting "Library" for "My Items" / object tree).

**Comment by unlikelyzero** [comment]
Ok cool. In that case, we usually lower the severity of bugs if it's been this way for a while

**Comment by davetsay** [comment]
@ozyx , is this a duplicate of #2964?

**Comment by ozyx** [comment]
> @ozyx , is this a duplicate of #2964?

I don't think so. And I'm actually unable to reproduce that one, so it may be fixed? This issue is moreso a side-effect of how we've implemented transactions, where deleting other objects from the tree while in edit mode tosses those into the same transaction for the focused object. SO when we cancel the transaction, they are restored.
